include/taskolib/Timeout.h

```diff
@@ -0,0 +1,183 @@
+/**
+ * \file   Timeout.h
+ * \author Lars Froehlich
+ * \date   Created on November 28, 2022
+ * \brief  Declaration of the Timeout class.
+ *
+ * \copyright Copyright 2022 Deutsches Elektronen-Synchrotron (DESY), Hamburg
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation, either version 2.1 of the license, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+#ifndef TASKOLIB_TIMEOUT_H_
+#define TASKOLIB_TIMEOUT_H_
+
+#include <chrono>
+#include <limits>
+
+#include "taskolib/exceptions.h"
+
+namespace task {
+
+/**
+ * A type for storing a timeout duration.
+ *
+ * A timeout can be zero, positive, or infinite. A default-constructed Timeout is
+ * infinite. The member functions is_finite() and is_infinite() can be used to test for
+ * infinity:
+ * \code
+ * using namespace std::literals;
+ *
+ * auto a = Timeout{ 1min }; // 1 minute via std::chrono duration parameter
+ * assert(a.is_finite());
+ *
+ * auto b = Timeout{ 60.0 }; // 60 seconds via floating-point parameter
+ * assert(b.is_finite());
+ *
+ * auto c = Timeout{}; // default-constructed Timeout is infinite
+ * assert(c.is_infinite());
+ *
+ * auto d = Timeout::infinity(); // static member function for generating infinite timeouts
```

Comment:
I agree that it is redundant. I just thought it might express the intent better. With `timeout = Timeout{}` you have to know something about the class whereas `timeout = Timeout::infinity()` is self-explanatory. But I don't have a strong opinion on that; we might as well remove it to keep the class tidier.

Reply:
see below